Ginger/GingerCoreNET/Run/RunSetActions/RunSetActionHTMLReport.cs

```diff
@@ -111,43 +111,81 @@ public override void Execute(ReportInfo RI)
                     GingerRunner gr = new GingerRunner();
                     runSetFolder = gr.ExecutionLoggerManager.GetRunSetLastExecutionLogFolderOffline();                    
                 }
-                if (!string.IsNullOrEmpty(selectedHTMLReportTemplateID.ToString()))
+
+                if (amdocs.ginger.GingerCoreNET.WorkSpace.Instance.RunningInExecutionMode)
                 {
-                    ObservableList<HTMLReportConfiguration> HTMLReportConfigurations = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>();
+                    string htmlReportFolderPath = "";
                     if (isHTMLReportFolderNameUsed && !String.IsNullOrEmpty(HTMLReportFolderName))
                     {
-                        string currentHTMLFolderName = string.Empty;
-                        if (!isHTMLReportPermanentFolderNameUsed)
+                        htmlReportFolderPath = Path.Combine(HTMLReportFolderNameCalculated, System.IO.Path.GetFileName(runSetFolder));
+                    }
+                    else if (!string.IsNullOrEmpty(WorkSpace.Instance.TestArtifactsFolder))
+                    {
+                        htmlReportFolderPath = Path.Combine(WorkSpace.Instance.TestArtifactsFolder, System.IO.Path.GetFileName(runSetFolder));
+                    }
+                    else
+                    {
+                        htmlReportFolderPath = Path.Combine(WorkSpace.Instance.SolutionRepository.ConvertSolutionRelativePath(ExtensionMethods.GetReportDirectory(WorkSpace.Instance.Solution.LoggerConfigurations.ExecutionLoggerConfigurationHTMLReportsFolder)), System.IO.Path.GetFileName(runSetFolder));
+                    }
+                    if (!string.IsNullOrEmpty(selectedHTMLReportTemplateID.ToString()))
+                    {
+                        ObservableList<HTMLReportConfiguration> HTMLReportConfigurations = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>();
+                        reportsResultFolder = Ginger.Reports.GingerExecutionReport.ExtensionMethods.CreateGingerExecutionReport(new ReportInfo(runSetFolder),
```

Comment:
why you duplicated also the code lines which do the actual report creation? can't you only modify the path it will be created under, set only param controling it instead have duplicated lines like 133 & 165?